File: UI/src/lib/types/cycle.ts
export type CycleStatus = 'upcoming' | 'active' | 'completed';

export interface CycleProgress {
	total: number;
	completed: number;
	cancelled: number;
}

export interface Cycle {
	id: string;
	team_id: string;
	name: string;
	number: number;
	status: CycleStatus;
	description: string | null;
	goals: string | null;
	retrospective: string | null;
	start_date: string | null;
	end_date: string | null;
	completed_at: string | null;
	progress?: CycleProgress;
	created_at: string;
	updated_at: string;
}

export interface CreateCycleRequest {
	name: string;
	description?: string;
	goals?: string;
	start_date: string;
	end_date: string;
}

export interface CycleBurndownPoint {
	date: string;
	scope: number;
	started: number;
	completed: number;
}

export interface UpdateCycleRequest {
	name?: string;
	description?: string;
	goals?: string;
	retrospective?: string;
	status?: CycleStatus;
	start_date?: string;
	end_date?: string;
}

export interface CompleteCycleRequest {
	retrospective?: string;
	carry_over?: boolean;
}

export interface VelocityPoint {
	cycle_id: string;
	cycle_name: string;
	cycle_number: number;
	scope: number;
	completed: number;
	cancelled: number;
	start_date: string | null;
	end_date: string | null;
}

File: BE/internal/dto/cycle.go
package dto

import "time"

type CreateCycleRequest struct {
	Name        string  `json:"name" validate:"required,min=1,max=100"`
	Description *string `json:"description"`
	StartDate   *string `json:"start_date"`
	EndDate     *string `json:"end_date"`
}

type UpdateCycleRequest struct {
	Name        *string `json:"name" validate:"omitempty,min=1,max=100"`
	Description *string `json:"description"`
	Status      *string `json:"status" validate:"omitempty,oneof=upcoming active completed"`
	StartDate   *string `json:"start_date"`
	EndDate     *string `json:"end_date"`
}

type CycleResponse struct {
	ID          string     `json:"id"`
	TeamID      string     `json:"team_id"`
	Name        string     `json:"name"`
	Number      int        `json:"number"`
	Status      string     `json:"status"`
	Description *string    `json:"description"`
	StartDate   *time.Time `json:"start_date"`
	EndDate     *time.Time `json:"end_date"`
	CompletedAt *time.Time `json:"completed_at"`
	Progress    *CycleProgressResponse `json:"progress,omitempty"`
	CreatedAt   time.Time  `json:"created_at"`
	UpdatedAt   time.Time  `json:"updated_at"`
}

type CycleProgressResponse struct {
	Total     int `json:"total"`
	Completed int `json:"completed"`
	Cancelled int `json:"cancelled"`
}
